protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,315 @@
+    - In the above example, such delta implementation will see version 7 as the latest snapshot.
+2. A client that understands managed commits can supplement the file listing result by contacting the commit-owner for information about un-backfilled commits, in order to get the most
+   recent snapshot of the table.
+    - In the above example, a delta implementation could get information about versions 0 through 7 from _delta_log directory and get information about un-backfilled commits (v8, v9) from the commit-owner.
+
+## Maintenance operations on managed-commit tables
+
+Maintenance operations such as [checkpoints](#checkpoints-1), [log compaction](#log-compaction-files), [metadata cleanup](#metadata-cleanup) can be performed only on the backfilled part of a managed-commit table.
+
+- A [checkpoint](#checkpoints-1) for a version `v` can be created only if the commit `v` is already backfilled i.e. `_delta_log/<v>.json` exists.
+- A [log compaction](#log-compaction-files) can be done only on the backfilled part of the table. The log compaction file `<x>.<y>.compacted.json` can be created only if commit `y` is already backfilled i.e. `_delta_log/<y>.json` exists.
+- A [Metadata cleanup](#metadata-cleanup) must always preserve the newest k >= 1 backfilled commits. In other words, [metadata cleanup](#metadata-cleanup) must always choose a backfilled `cutOffCheckpoint`.
+    - This allows the table to remain discoverable by the filesystem-based delta readers even after metadata cleanup deletes everything before commit version `cutOffCheckpoint`.
+
+## Managed Commit Enablement
+
+The managed-commit feature is supported and active when:
+- The table must be on Writer Version 7.
+- The table has a `protocol` action with `writerFeatures` containing the feature `managedCommits`.
+- The table has a metadata property `delta.managedCommitOwnerName` in the [change-metadata](#change-metadata)'s configuration.
+- The table may have a metadata property `delta.managedCommitOwnerConf` in the [change-metadata](#change-metadata)'s configuration. The value of this property is a json-coded string-to-string map.
+    - A commit-owner can store additional information (e.g. configuration information such as service endpoints) in this field, for use by the commit-owner client (it is opaque to the Delta client).
+
+Note that a table is in invalid state if the change-metadata contains the `delta.managedCommitOwnerName` property but the table does not have the `managedCommits` feature in the `protocol` action or vice versa.
+
+E.g.
+```json
+{
+  "metaData":{
+    "id":"af23c9d7-fff1-4a5a-a2c8-55c59bd782aa",
+    "format":{"provider":"parquet","options":{}},
+    "schemaString":"...",
+    "partitionColumns":[],
+    "configuration":{
+      "appendOnly": "true",
+      "delta.managedCommitOwnerName": "commit-owner-1",
+      "delta.managedCommitOwnerConf": "{\"batchSize\":\"10\",\"endpoint\":\"http://sample-url.com/commit\"}"
```

Comment:
```suggestion
      "delta.managedCommitOwnerConf": "{\"endpoint\":\"http://sample-url.com/commit\", \"authenticationMode\":\"oauth2\"}"
```
(no idea what "batchSize" might mean, and probably not worth the confusion just for an example)

Reply:
Also, somewhere we should say that the commit owner conf should never include secrets such as auth tokens or credentials, because any reader with access to the table's storage location can see them.

Reply:
Added this in `Managed Commit Enablement` section where we describe the table property.